File: app/src/main/java/com/compuspulse/data/remote/model/PageResponse.java
package com.compuspulse.data.remote.model;

import java.util.ArrayList;
import java.util.List;

public class PageResponse<T> {
    private int curPage;
    private List<T> datas;
    private int offset;
    private boolean over;
    private int pageCount;
    private int size;
    private int total;

    public int getCurPage() {
        return curPage;
    }

    public List<T> getDatas() {
        return datas == null ? new ArrayList<>() : datas;
    }

    public int getOffset() {
        return offset;
    }

    public boolean isOver() {
        return over;
    }

    public int getPageCount() {
        return pageCount;
    }

    public int getSize() {
        return size;
    }

    public int getTotal() {
        return total;
    }
}

File: app/src/main/java/com/compuspulse/data/remote/model/PageResponse.kt
package com.compuspulse.data.remote.model

class PageResponse<T> {
    private var curPage: Int = 0
    private var datas: List<T>? = null
    private var offset: Int = 0
    private var over: Boolean = false
    private var pageCount: Int = 0
    private var size: Int = 0
    private var total: Int = 0

    fun getCurPage(): Int = curPage

    fun getDatas(): List<T> = datas ?: ArrayList()

    fun getOffset(): Int = offset

    fun isOver(): Boolean = over

    fun getPageCount(): Int = pageCount

    fun getSize(): Int = size

    fun getTotal(): Int = total
}
